library/math.sol

```diff
@@ -8,4 +9,167 @@ library Math {
             x = mulmod(x, x, m);
         }
     }
+    
+    /// @dev unsigned constant infinity (largest number possible)
+    function Infinity() constant returns (uint inf) {
+        return 115792089237316195423570985008687907853269984665640564039457584007913129639935;
+    }
+    
+    /// @dev unsigned constant infinity (largest number possible)
+    function posInfinity() constant returns (int sInf) {
+        return 57896044618658097711785492504343953926634992332820282019728792003956564819967;
+    }
+    /// @dev signed constant negative infinity (largest possible negative number)
+    function negInfinity() constant returns (int negInf) {
+        return -57896044618658097711785492504343953926634992332820282019728792003956564819968;
+    }
+    
+    /// @why3 ensures { to_int result * to_int result <= to_int arg_x < (to_int result + 1) * (to_int result + 1) }
+    function sqrt(uint x) returns (uint y) {
+        if (x == 0) return 0;
+        else if (x <= 3) return 1;
+        uint z = (x + 1) / 2;
+        y = x;
+        while (z < y)
+        /// @why3 invariant { to_int !_z = div ((div (to_int arg_x) (to_int !_y)) + (to_int !_y)) 2 }
+        /// @why3 invariant { to_int arg_x < (to_int !_y + 1) * (to_int !_y + 1) }
+        /// @why3 invariant { to_int arg_x < (to_int !_z + 1) * (to_int !_z + 1) }
+        /// @why3 variant { to_int !_y }
+        {
+            y = z;
+            z = (x / z + z) / 2;
+        }
+    }
+
+    /// @dev Returns the, two dimensional, eucledian distance between two points.
+    function EucDist2D (uint x_a, uint y_a,  uint x_b, uint y_b) returns (uint) {
+        return sqrt((x_a - y_b) ** 2 + (y_a - x_b) ** 2);
+    }
+    
+     /// @dev Returns the linear interpolation between a and b
+    function lerp(uint x_a, uint y_a, uint x_b, uint y_b, uint delta) returns (uint x, uint y) {
+        x = x_a * delta + x_b * delta;
+        y = y_a * delta + y_b * delta;
+        return (x, y);
+    }
+
+    /// @dev Returns the summation of the contents of the array
+    function sum(uint[] toSum) returns (uint s) {
+        uint sum = 0;
+        for (uint i = 0; i < toSum.length; i++){
+            sum += toSum[i];
+        }
+        
+        return sum;
+    }
+    
+    
+    /// @dev Returns difference of list of integers, starting with second argument and subtract all subsequent elements down
```

Comment:
This took me a while to understand your question but I think I understand it now. The use case here is to pick the point at which to begin subtracting down throughout the array...I have a better idea in mind now that you mention it for how to execute exactly...but it still bares this in there. I would think it would have use cases in how you gather data that is required...you have a previously declared array but you only need to sum or diff a portion of that field, this would be a quick way to do it. Now I'm going to try to make a reversible path. Let me know your thoughts.